Lib/asyncio/base_events.py

```diff
@@ -601,29 +601,62 @@ def _check_running(self):
             raise RuntimeError(
                 'Cannot run the event loop while another loop is running')
 
-    def run_forever(self):
-        """Run until stop() is called."""
+    def run_forever_setup(self):
+        """Set up an event loop so that it is ready to start actively looping
+        to process events.
+
+        Returns the state that must be restored when the loop concludes. This state
+        should be passed in as arguments to ``run_forever_cleanup()``.
```

Comment:
Document the type of the state, please. (If it's meant to be opaque, make it more opaque than a 1-tuple please. :-)

I find it a bit odd that the method name is "run_forever_setup()", which doesn't hint at the existence of a return value. (Could you store the state to be restored in a private/protected instance variable instead of returning it?)

Also, "arguments"? Or "argument"?

Reply:
I've modified the implementation to store the state as a protected variable, which simplifies the interface.